components/sup/src/manager/service/package.rs

```diff
@@ -101,6 +101,8 @@ pub struct Pkg {
     pub svc_run: PathBuf,
     pub svc_user: String,
     pub svc_group: String,
+    pub user_path: PathBuf,
```

Comment:
I'm not sure if adding new fields here create some serialization problem. Like during the update of supervisor - we serialize this struct, store it on the disk, update the supervisor, deserialize the struct from the disk and we get empty fields for `user_path` and `user_config_path`. Is this something to be defensive about (by detecting if the fields are empty, for instance)?

Reply:
Hrm... that's a good question. I wonder if these really need to be public members of the `Pkg` struct at all; maybe they should just stay function calls (which you use with your `PackageForConfig` trait). You could just use the `fs::user_path(&package.ident.name)`, etc. implementations directly in your trait implementation for `Pkg`, get the exact same behavior, and avoid any serialization issues.